src/Cmfd.cpp

```diff
@@ -2627,9 +2635,12 @@ CMFD_PRECISION Cmfd::getUpdateRatio(int cell_id, int group, int fsr) {
  */
 CMFD_PRECISION Cmfd::getFluxRatio(int cell_id, int group, int fsr) {
 
-  double* interpolants = _axial_interpolants.at(fsr);
+  double* interpolants;
   double ratio = 1.0;
-  if (interpolants[0] != 0 || interpolants[2] != 0) {
+  if (_k_nearest > 0)
```

Comment:
Actually _use_axial_interpolation and _centroid_update_on are independent options for seperately axial and radial directions. And _centroid_update_on=false should be equivalent to _centroid_update_on=true and _knearest=1.
I have fixed some bugs here, which we may have a discussion on.